src/wasm-type.h

```diff
@@ -576,6 +576,13 @@ struct TypeBuilder {
   void setHeapType(size_t i, Struct&& struct_);
   void setHeapType(size_t i, Array array);
 
+  // This is an ugly hack around the fact that temp heap types initialized with
+  // BasicHeapTypes are not themselves considered basic, so `HeapType::isBasic`
+  // and `HeapType::getBasic` do not work as expected with them. Call these
+  // methods instead.
+  bool isBasic(size_t i);
+  HeapType::BasicHeapType getBasic(size_t i);
```

Comment:
Is fixing this issue not practical? I've run into it too and it is annoying.

Reply:
`isBasic` is currently a single integer comparison that does not need to know any of the implementation details in `wasm-type.cpp`, but fixing this would require making it more complicated. I should probably check whether the performance difference is noticeable, and if it's not, just fix this the "right" way.